micropip/_compat/_compat_not_in_pyodide.py

```diff
@@ -52,14 +52,29 @@ async def fetch_string_and_headers(
         return response.read().decode(), headers
 
     @staticmethod
-    def get_dynlibs(archive: IO[bytes], suffix: str, target_dir: Path) -> list[str]:
-        return []
-
-    @staticmethod
-    async def loadDynlibsFromPackage(
-        pkg_metadata: "PackageData", dynlibs: list[str]
+    async def install(
+        buffer: Any,
+        filename: str,
+        install_dir: str,
+        metadata: dict[str, str] | None = None,
     ) -> None:
-        pass
+        """
+        Install a package from a buffer to the specified directory.
+        TODO: Remove host tests that depends on internal behavior of install (https://github.com/pyodide/micropip/issues/210)
+              to make the compat code simpler
+        """
+        from micropip.metadata import wheel_dist_info_dir
+
+        with zipfile.ZipFile(io.BytesIO(buffer)) as zf:
+            zf.extractall(install_dir)
+            normalized_name = filename.split("-")[0]
+            dist_dir = Path(install_dir) / wheel_dist_info_dir(zf, normalized_name)
```

Comment:
Should the normalization happen as per https://packaging.python.org/en/latest/specifications/name-normalization/#name-normalization, rather than `split("-")[0]`?